lib/questions.rb

```diff
@@ -1,22 +1,30 @@
 # keep only the elements that start with an a
 def select_elements_starting_with_a(array)
+  array.select {|fruit| fruit[0] == "a"}
 end
 
 # keep only the elements that start with a vowel
 def select_elements_starting_with_vowel(array)
+  array.select {|name| name[0].match(/[aeiou]/)}
 end
 
 # remove instances of nil (but NOT false) from an array
 def remove_nils_from_array(array)
+  array.reject {|letter| letter == nil}
+  # there's also "array.compact". Neat.
 end
 
 # remove instances of nil AND false from an array
 def remove_nils_and_false_from_array(array)
+  # could have array.compact before 
+  # and just reject those with false in the block below
+  array.reject{|letter| letter == nil || letter == false}
 end
 
 # don't reverse the array, but reverse every word inside it. e.g.
 # ['dog', 'monkey'] becomes ['god', 'yeknom']
 def reverse_every_element_in_array(array)
+  array.each {|animal| animal.reverse! }
```

Comment:
Space between { and | missing.<br>Pass `&:reverse!` as an argument to `each` instead of a block.